diff --git a/src/linux/fs.cpp b/src/linux/fs.cpp
index 306beab..4f4e9e3 100644
--- a/src/linux/fs.cpp
+++ b/src/linux/fs.cpp
@@ -439,6 +439,16 @@ Try<Nothing> pivot_root(
   // 'pivot_root' on the x86_64 architecture, see
   // arch/x86/syscalls/syscall_64.tbl
   int ret = ::syscall(155, newRoot.c_str(), putOld.c_str());
++#elif __thumb__
+  // A workaround for arm thumb mode. The magic number '218' is the syscall
+  // number for 'pivot_root' on the arm thumb mode, see
+  // https://w3challs.com/syscalls/?arch=arm_thumb
+  int ret = ::syscall(218, newRoot.c_str(), putOld.c_str());
+#elif __arm__
+  // A workaround for arm. The magic number '9437402' is the syscall
+  // number for 'pivot_root' on the arm architecture, see
+  // https://w3challs.com/syscalls/?arch=arm_strong
+  int ret = ::syscall(9437402, newRoot.c_str(), putOld.c_str());
 #else
 #error "pivot_root is not available"
 #endif
